### ISSUE TITLE
Bloodborne freezes often when moving around, requiring constant debugger pause/resume

### ISSUE THREAD

**Issue by C0rn3j** [shadps4-emu/shadPS4#1792]
Issue originally posted in this PR - https://github.com/shadps4-emu/shadPS4/pull/1777

The game keeps freezing (not crashing), and the only way to get it unstuck is to have it launched in a debugger and constantly pause and unpause whenever it happens.

When userfaultfd is enabled, it happens immediately upon entering the character creator.

When it is disabled, it triggers later during loading screens or more often when running around in the game - usually within a minute or less of gameplay.

Full all-thread backtraces from when the game is frozen are available in here - https://github.com/shadps4-emu/shadPS4/pull/1777#issuecomment-2543111947 and here https://github.com/shadps4-emu/shadPS4/pull/1777#issuecomment-2543132430

Log from a build around some Dec 8: [shad_log.txt](https://github.com/user-attachments/files/18053535/shad_log.txt) - it happens on a build from today too, though.

**Comment by Missake212** [comment]
Hey, I was wondering if your PR #1777  also "solves" this issue ?

**Comment by C0rn3j** [comment]
Nope, it just makes it happen half a minute later.

**Comment by C0rn3j** [comment]
Having said that, I tested with latest commits and I couldn't reproduce in 2 minutes, and it was very very consistent before, so *maybe* it is gone now with the userfd patch + recent patches combined, will try later.

**Comment by C0rn3j** [comment]
I cannot repro on https://github.com/shadps4-emu/shadPS4/commit/43fd8fa44a4ea7e8bdfba86106e72e57d7e678d5, I can with `-DENABLE_USERFAULTFD=ON`

I can repro from an older build I have hanging around https://github.com/shadps4-emu/shadPS4/commit/e752f04cde25941818e141cc933ffa380cc321e5 and I have two versions, I believe one with userfd patch and one without, and both are capable of failing even on the loading screen from loading a savefile.

So it looks like the userfaultfd is causing freezes, and some other issue which was fixed in the last couple days (I tested somewhat often, so it was probably fixed in the last 24h) was too.

TL;DR there were two issues. Latest master + disabled userfd works, enabled userfd has freezes.
